lib/internal/Magento/Framework/Code/Reader/ArgumentsReader.php

```diff
@@ -98,8 +102,8 @@ public function getConstructorArguments(\ReflectionClass $class, $groupByPositio
      */
     private function processType(\ReflectionClass $class, \Laminas\Code\Reflection\ParameterReflection $parameter)
     {
-        if ($parameter->getClass()) {
-            return NamespaceResolver::NS_SEPARATOR . $parameter->getClass()->getName();
+        if ($parameterClass = $this->getParameterClass($parameter)) {
```

Comment:
It's better to avoid assignment in coditions to avoid common future mistakes related to this construction.
```suggestion
        $parameterClass = $this->getParameterClass($parameter)
        if (parameterClass) {
```